---
title: "Project 2"
output: html_notebook
---
Scientific Question: Do long-lived animal species have a more similar sequence to human FOXO3 (a gene known to have effects on longevity) than short-lived species, and does FOXO3 play a significant role in longevity of a long versus short lived species?

FOXO3 is a stress-related gene that has been found to be a major gene for human longevity.  It belongs to the class O of Forkhead transcription factors, which participates in a wide range of cellular processes such as apoptosis, metabolism, and cell cycle arrest (Sun 2017). FOXO3 has been found to be very strongly associated with long-lived animals. It is a very highly conserved gene that is known to regulate IGF1 gene.  IGF1 is a gene that is recognized to regulate growth and aging, and it has been found that when IGF1 signals are reduced, the lifespan of an organism increases (Kim 2020).  Adding on, FOXO3 is correlated with Daf-16, which is also lifespan extender gene.   Lastly, FOXO3 is a direct target of p53, where p53 transactivates the expression of FOXO3, which in turn modulates p53 dependent apoptosis.  Under stress, it has been found that p53 inhibits FOXO3 transcriptional activity, and this in turn decreases FOXO3 expression levels, and results in a decrease of longevity.  What happens is that p53 is recruited to a binding site in the second intron of the FOXO3 gene,and this is where SNPs associated with extreme human longevity are found. It has been found that FOXO3 is directly correlated to p53 tumor suppressor (Renault 2012). 

Thus, since FOXO3  is a gene known to be involved with longevity, we are comparing the FOXO3 gene sequences between long and short lived animals to humans, to see whether or not FOXO3 is playing a significant role in longevity. 

Scientific Hypothesis: If long-lived animals and short-lived animals have different FOXO3 gene sequences, then humans and chimps (a long-lived species) will have a more similar sequence than humans and mice (a short-lived species).

Here we perform BLAST, multiple sequence alignment, and a dot plot to show the similarity between the three different subjects being tested. First, BLAST was done to find regions of local similarity between nucleotide sequences of other FOXO3 genes in different organisms. BLAST was performed through NCBI, on the homo sapiens (human) FOXO3 gene (accession 3 NM_001455.4), which was my focal sequence. Once a BLAST search was made, I wanted to compare the human FOXO3 gene with a long-lived and short-lived animal.  So, looking through significant alignments with the human FOXO3, I found alignments with a Chimpanzee (pantroglodytes), which is long-lived, and a Chinese mouse (Cricetulus griseus), which is short-lived... both of their e-values were 0.0. Next, after I selected my two other organisms that have high similarity with the human FOXO3 gene, I did a multiple sequence alignment. MSA was used to compare all the sequences together (human, Chimp, and mouse), to see which of the two animals, the Chimpanzee or mouse, had a higher similarity to the human FOXO3 gene sequence. Lastly, a dot plot was performed.  Two dot plots were done, one showing the similarity between human and chimp, and one showing the similarity between human and mouse.  This was done to allow me to know the relationship between the two animal FOXO3 gene sequence,with that of the humans.  The more dense one plot is than the other, indicates that that animal has a more similar sequence to the human sequence, and thus illustrating whether long or short - lived species have a more similar sequence to the human FOXO3 gene. The length of the chimpanzee sequence is 8176, the length for the mouse is 6531, and the human length is 7296.  A global variable is a variable that you are able to access globally throughout the program.  A local variable is a variable that can only be accessed within a particular block of code or the function that it is defined in.  In my project, while there are no local variables, there are multiple global variables which you will see labeled throughout.

Below, this bit of code is responsible for loading in all of the necessary packages. Here is where I will define the packages and what they are loaded in for.

1) Biostrings: Biostrings is a package that is composed of classes and functions that represent biological strings such as DNA, RNA, and amino acids. It also can function for performing short read alignments, and pairwise sequence alignments.  Biostrings also functions for reading and writing outputs such as FASTA files. It was loaded in here to load in the FASTA file for the human FOXO3 gene sequences, to allow for a BLAST search to be made.  The function being used here is readDNAStringSet and toString.  readDNAStringSet reads in the FASTA file and toString sets the FASTA file to the variable 'Human'.  https://bioconductor.org/packages/release/bioc/html/Biostrings.html  
2) annotate:  Annotate is a package that is used to create an annotation layer.It also added geometry to the plot that is formed for the BLAST search.  It is loaded in here because it provides an explanation for the function 'BlastSeqeunces', and allows for the DNA to be aligned neatly. The function being used for annotate is blastSequences. This functuon is used to allow us to perform a BLAST on the human FOXO3 gene sequence.  https://www.rdocumentation.org/packages/ggplot2/versions/3.3.5/topics/annotate
3) msa:  MSA is a package that provides Bioconducter interface to perform a multiple sequencing alignment through particular algorithms. There are three multiple sequencing alignment algorithms in this package: ‘ClustalW’, ‘ClustalOmega’, and ‘MUSCLE’.It is loaded in for to perform the multiple sequence alignment between the human, chimp, and mouse FOXO3 gene sequences. readDNAStringSet and myFirstAlignment are used as the functions from the msa package. http://bioconductor.org/packages/release/bioc/html/msa.html
4) seqinr:  It is a package that allows you to retrieve and analyze biological sequences. It provides direct access to protein coding regions, rRNA coding regions, etc. that are available in GenBank.  Here it was used to retrieve the sequences from the FASTA files to help R form the dot plot accurately.The function following the installation of seqinr was 'dotPlot', and seqinr was required for this function to work because the sequenced of the FOXO3 genes needed to be retrieved from the FASTA files. https://cran.r-project.org/web/packages/seqinr/index.html
```{r}
#Install the packages using these commands 
library(Biostrings)
library(annotate)
library(msa)
library(seqinr)
```


While BLAST was done through NCBI, another method to do BLAST (yet more difficult to interpret), is done below.  This code is responsible for loading in the FASTA file for the Human FOXO3 gene sequence.  This was done using readDNAStringSet, which directly allows access to the subset of sequences from the fasta file.  Then, BLAST hit was performed on the sequence. 
#BLAST
```{r}
#install.packages("Biostrings")
library(Biostrings)

# Read in FOXO3human fasta -> which is the sequence of the human FOXO3 gene
Human <- toString(readDNAStringSet("~/Downloads/FOXO3human.fasta", format = "fasta", nrec = 1L, skip = 0L, seek.first.rec = FALSE, use.names = TRUE, with.qualities = FALSE ))

#print variable for error/quality check
print(Human)

#install.packages("annotate")
#in order for this code to work,when you install "annotate", make sure to type 'n' when it asks you to update all/some/none
BiocManager::install("annotate")
library(annotate)

#perform the BLAST on the human FOXO3 gene sequence... it may take around a total of 250 seconds, make sure to type 'y' every time it asks if you should (wait another 40 seconds?)
blastSequences(Human)
```
The above fasta file here was searched through NCBI.  It consists of the homo sapiens transcript variant 1.

In the next line of code, I used msa to compare the sequences of all three FOXO3 sequences, the human, chimp, and mouse. This will be used to recognize the similarities and differences between the nucleotides in these sequences. 
#Multiple Sequence Alignment
```{r}
if ( !requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("msa")

#install.packages("msa")
library(msa)
#Read in the FASTA file that contains the FOXO3 gene sequences of the human, mouse, and chimpanzee. mySeqeunces is a global variable 
#make sure to combine all 3 different FOXO3 sequences into one file to ensure that the msa runs properly 
mySequences <- readDNAStringSet("FOXO3all.fasta")
mySequences
#perform a multiple sequence alignment of the DNAStringSet that contains all 3 sequences
#since mySequences is a global variable, myFirstAlignmnet can be used to align the sequences in the fasta file. 
myFirstAlignment <- msa(mySequences)

```

In the next 2 blocks of code, three FASTA files are loaded in (the human, mouse, and chimp FOXO3 gene sequences). This is done to follow up with the formation of two dot plots.  One showing the significance of similarity between the human and mouse FOXO3 gene sequences, and one showing the similarity between the human and chimp FOXO3 gene sequence. 
#Dotplot
```{r}
#install.packages("seqinr")
install.packages("seqinr")
library("seqinr")

#read in the FASTA files that will be drawn in the dotplots, and print each sequence 

#FOXO3human is a global variable 
FOXO3human <- read.fasta("~/Downloads/FOXO3human.fasta")
FOXO3human

#FOXO3mice is a global variable 
FOXO3mice <- read.fasta("~/Downloads/FOXO3mice.fasta")
FOXO3mice

#perform 1st dotplot... between human and mice FOXO3 gene sequences. We will be changing the parameters of the dotplot in order to allow us better interpret it.  The window size will be 3, where we are looking at 3 nucleotides at a time,and the number of matches is also 3, which represents the number of matches to plot per window.
dotPlot(FOXO3human[[1]],FOXO3mice[[1]], wsize=4, nmatch=4, col=c("white", "black"))
```

```{r}
install.packages("seqinr")
library("seqinr")

#read in the FASTA files that will be drawn in the dotplots, and print each sequence 

#FOXO3human is a global variable 
FOXO3human <- read.fasta("~/Downloads/FOXO3human.fasta")
FOXO3human

#FOXO3Chimp is a global variable 
FOXO3Chimp <- read.fasta("~/Downloads/FOXO3chimp.fasta")
FOXO3Chimp

#perform the 2nd dot plot... between human and chimp FOXO3 gene sequences.  We will be changing the parameters of the dotplot in order to allow us better interpret it.  The window size will be 3, where we are looking at 3 nucleotides at a time,and the number of matches is also 3, which represents the number of matches to plot per window.
dotPlot(FOXO3human[[1]], FOXO3Chimp[[1]], wsize=4, nmatch=4, col=c("white", "black"))
```

Analysis of Results: 
The first dot plot illustrates the similarity of sequences between the FOXO3 gene in humans and mice, while the second dot plot shows the similarity between the human and the chimpanzee FOXO3 gene sequences.  While both the mouse and chimpanzee appear to be very similar to the human FOXO3 gene sequence, it appears that the mice has a more significant similarity with the human sequence, than the chimp.  This conclusion was drawn because when looking at the chimp versus human plot, it appears  that the sequences begin to match after the first 900 nucleotides of the FOXO3 Chimpanzee sequence.  On the other hand, the human versus mouse plot does not have any disruptions, and the sequences begin to match at the very beginning (at 0).
In both plots, there does not appear to be any major insertions or deletions since there are no disruptions in the diagonal line, and thus there does not appear to be any frame shifts in either plots. Also, both plots have lots of dots across the diagonal line which indicates that the two protein sequences being compared in each plot contain many identical amino acids at the same (or very similar) positions along their lengths.  Therefore, the short-lived Chinese mouse had a more similar FOXO3 sequence to the humans sequence.  Thus, my hypothesis was not supported, having a similar sequence to the human FOXO3 gene sequence does not determine whether an animal is long lived or short lived, and illustrates that there may be other factors involved in longevity, particularly for chimps, than solely FOXO3.    


References: 
- Sun X, Chen WD, Wang YD. DAF-16/FOXO Transcription Factor in Aging and Longevity. Front Pharmacol. 2017 Aug 23;8:548. doi: 10.3389/fphar.2017.00548. PMID: 28878670; PMCID: PMC5572328.
- Kim, Sang-Eun et al. “Does Calorie Restriction Modulate Inflammaging via FoxO Transcription Factors?.” Nutrients vol. 12,7 1959. 30 Jun. 2020, doi:10.3390/nu12071959
- Renault, V M et al. “The pro-longevity gene FoxO3 is a direct target of the p53 tumor suppressor.” Oncogene vol. 30,29 (2011): 3207-21. doi:10.1038/onc.2011.35
